corehq/apps/reports/util.py

```diff
@@ -125,6 +128,18 @@ def possible_usernames():
         return HQUserType.human_readable[HQUserType.ADMIN]
 
 
+def get_user_id(form_id):
+    key = f'xform-{form_id}-user_id'
+    user_id = cache.get(key)
+    if not user_id:
+        try:
+            user_id = FormAccessorSQL.get_form(form_id).metadata.userID
```

Comment:
SQL forms have a `user_id` field which can be used instead of accessing the metadata (which requires loading the full XML etc).

```suggestion
            user_id = FormAccessorSQL.get_form(form_id).user_id
```

Reply:
https://github.com/dimagi/commcare-hq/pull/30965/commits/a3df1bf86e1625e3e33d5a31a9f185e5b4208f08